src/sonic-db-cli/sonic-db-cli.cpp

```diff
@@ -0,0 +1,239 @@
+#include <execution>
+#include <iostream>
+
+#include <swss/dbconnector.h>
+#include <swss/sonicv2connector.h>
+
+#include "sonic-db-cli.h"
+
+static std::string name_space;
+static std::string db_or_operation;
+const char* empty_str = "";
+
+void printUsage(const po::options_description &all_options)
+{
+    std::cout << "usage: sonic-db-cli [-h] [-s] [-n NAMESPACE] db_or_op [cmd ...]" << std::endl;
+    std::cout << all_options << std::endl;
+    
+    std::cout << "**sudo** needed for commands accesing a different namespace [-n], or using unixsocket connection [-s]" << std::endl;
+    std::cout << "" << std::endl;
+    std::cout << "Example 1: sonic-db-cli -n asic0 CONFIG_DB keys \\*" << std::endl;
+    std::cout << "Example 2: sonic-db-cli -n asic2 APPL_DB HGETALL VLAN_TABLE:Vlan10" << std::endl;
+    std::cout << "Example 3: sonic-db-cli APPL_DB HGET VLAN_TABLE:Vlan10 mtu" << std::endl;
+    std::cout << "Example 4: sonic-db-cli -n asic3 APPL_DB EVAL \"return {KEYS[1],KEYS[2],ARGV[1],ARGV[2]}\" 2 k1 k2 v1 v2" << std::endl;
+    std::cout << "Example 5: sonic-db-cli PING | sonic-db-cli -s PING" << std::endl;
+    std::cout << "Example 6: sonic-db-cli SAVE | sonic-db-cli -s SAVE" << std::endl;
+    std::cout << "Example 7: sonic-db-cli FLUSHALL | sonic-db-cli -s FLUSHALL" << std::endl;
+}
+
+void printRedisReply(redisReply* reply)
+{
+    switch(reply->type)
+    {
+    case REDIS_REPLY_INTEGER:
+        std::cout << reply->integer;
+        break;
+    case REDIS_REPLY_STRING:
+    case REDIS_REPLY_ERROR:
+    case REDIS_REPLY_STATUS:
+    case REDIS_REPLY_NIL:
+        std::cout << std::string(reply->str, reply->len);
+        break;
+    case REDIS_REPLY_ARRAY: 
+        for (size_t i = 0; i < reply->elements; i++)
+        {
+            printRedisReply(reply->element[i]);
+        }
+        break;
+    default:
+        std::cerr << reply->type << std::endl;
+        throw std::runtime_error("Unexpected reply type");
+    }
+
+    std::cout << std::endl;
+}
+
+int executeCommands(
+    const std::string& db_name,
+    std::vector<std::string>& commands,
+    const std::string& name_space,
+    bool use_unix_socket)
+{
+    std::unique_ptr<swss::SonicV2Connector_Native> dbconn;
+    if (name_space.compare("None") == 0) {
+        dbconn = std::make_unique<swss::SonicV2Connector_Native>(use_unix_socket, empty_str);
```

Comment:
>SonicV2Connector_Native [](http://example.com/codeflow?start=40&length=23)

The classes of SonicV2Connector* are only used by SWIG and then by python application. The equal class in C++ is DBConnector.